web-registry/src/components/organisms/ChooseCreditClass/hooks/useGetCreditClassOptions.tsx

```diff
@@ -0,0 +1,82 @@
+import { useState, useEffect } from 'react';
+
+import useQueryListClassesWithMetadata from '../../../../hooks/useQueryListClassesWithMetadata';
+import { client } from '../../../../sanity';
+import { useAllCreditClassQuery } from '../../../../generated/sanity-graphql';
+import { useAllCreditClassesQuery } from '../../../../generated/graphql';
+import { useWallet } from '../../../../lib/wallet';
+
+interface CreditClassOption {
+  id: string;
+  onChainId: string;
+  imageSrc?: string;
+  title: string;
+  description?: string;
+}
+
+function useGetCreditClassOptions(): {
+  creditClassOptions: CreditClassOption[];
+  loading: boolean;
+} {
+  const [loading, setLoading] = useState(true);
+  const [creditClassOptions, setCreditClassOptions] = useState<
+    CreditClassOption[]
+  >([]);
+  const { wallet } = useWallet();
+  const onChainClasses = useQueryListClassesWithMetadata();
+  const { data: offChainCreditClasses } = useAllCreditClassesQuery();
+  const { data: creditClassContentData } = useAllCreditClassQuery({ client });
+
+  useEffect(() => {
+    if (!wallet?.address || onChainClasses?.length < 1) return;
+
+    const issuerCreditClasses = onChainClasses?.filter(
+      onChainClass => onChainClass.issuers?.indexOf(wallet.address) > -1,
```

Comment:
This is equivalent but maybe a bit more readable?

```suggestion
      onChainClass => onChainClass.issuers?.includes(wallet.address),
```

Reply:
sounds good